tests/core/pthread/test_pthread_dlopen.c

```diff
@@ -24,14 +24,17 @@ static void* thread_main() {
   // Spin until the main thread has loaded the side module
   while (!ready) {}
 
-  // Without this explict sync we get "invalid index into function table" below
-  _emscripten_thread_sync_code();
+  // Without this explict yield we could "invalid index into function table"
+  // below because this thread might not have loaded the side module.
+  // Uncommend the calls to printf will also, in practice, end up loading
+  // the side module because internally they may end up wait on a lock.
```

Comment:
Ah, so we want to test that the load is caused by sched_yield and not something else; that makes sense. Is there a way to also check that the libraries are not loaded?

Reply:
I thought about that... bit that might be over constraining the test.   Maybe its worth it to test the failure case?

Reply:
If there is an easy or likely way that the test could accidentally pass (e.g. by adding something innocuous-looking like a printf), then IMO it might be worth checking that that doesn't happen.

Reply:
Yup, will do.   I guess its good to test for when we expect the code to not be loaded.

Reply:
OK, I updated to the test so that we verify it fails without the yield